sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java

```diff
@@ -124,14 +124,16 @@ private Combine() {
     return globally(fn, displayDataForFn(fn));
   }
 
-  private static <T> DisplayData.ItemSpec<? extends Class<?>> displayDataForFn(T fn) {
-    return DisplayData.item("combineFn", fn.getClass())
+  private static DisplayData.ItemSpecBase<?, ?> displayDataForFn(Object fn) {
```

Comment:
This pattern is still not great, and pervasive on `PTransforms` which take a user-fn from many different interfaces or adapt them for different cases. The conventional pattern goes:

1. On all public APIs which take a user-fn, immediately capture the display data for the input fn.
  * Since this happens on many public APIs, use a helper-method to capture the display data consistently.
  * In some cases the user-fn will implement `HasDisplayData`, and in some cases it won't-- wrap with either `DisplayData.item(..)` or `DisplayData.nested(..)` accordingly.
2. When adapting the user-fn, retail the original captured display data.

Ideally we should have a terse syntax for this common scenario. Some ideas:

1. Extend the notion of "inferring" the display data type and have one API that can register a nested sub-component or just the class.
2. Expand .`nested(..)` to take any `Object`, and only include the sub-component if it implements `HasDisplayData`
3. Create an easy syntax of wrapping any Object in a `HasDisplayData` noop-stub so `.nested()` can always be used.


Ideas welcome.